agent/checks/check.go

```diff
@@ -907,17 +908,19 @@ type StatusHandler struct {
 	logger                 hclog.Logger
 	successBeforePassing   int
 	successCounter         int
+	failuresBeforeWarning  int
 	failuresBeforeCritical int
 	failuresCounter        int
 }
 
 // NewStatusHandler set counters values to threshold in order to immediatly update status after first check.
-func NewStatusHandler(inner CheckNotifier, logger hclog.Logger, successBeforePassing, failuresBeforeCritical int) *StatusHandler {
+func NewStatusHandler(inner CheckNotifier, logger hclog.Logger, successBeforePassing, failuresBeforeWarning, failuresBeforeCritical int) *StatusHandler {
```

Comment:
Accepting so many ints as function arguments is likely going to be error prone (easy to get them mixed up), but I think that's something we can address later. It should be fine for now.

Reply:
Yes. I also thought that this might be a bit much, but I don't know the codebase that well so I didn't want to get started on any refactoring in this change. I guess we could do a data structure to pass those around, but that could be done separately, as it would probably complicate this change considerably.